File: src/data_structure/linked_list/insert_at_head.cpp
/*
  Insert Node at the begining of a linked list
  Initially head pointer argument could be NULL for empty list
  Node is defined as 
  struct Node
  {
     int data;
     struct Node *next;
  }
return back the pointer to the head of the linked list in the below method.
*/
Node* Insert(Node *head,int x)
{
    Node *n = (struct Node *)malloc(sizeof(struct Node));
    n->data = x;
    if(head == NULL)
    {
        n->next = NULL;
        return n;
    }
    n->next = head;
    return n;
}

File: src/data_structure/linked_list/insert_at_head.java
/*
  Insert Node at the end of a linked list 
  head pointer input could be NULL as well for empty list
  Node is defined as 
  class Node {
     int data;
     Node next;
  }
*/
// This is a "method-only" submission. 
// You only need to complete this method. 

Node Insert(Node head,int x) {
    Node n = new Node();
    n.data = x;
    if (head == null) {
        n.next = null;
        return n;
    }
    n.next = head;
    return n;
}

